libraries/SITL/SIM_FlightAxis.cpp

```diff
@@ -104,7 +104,11 @@ void *FlightAxis::update_thread(void *arg)
     FlightAxis *flightaxis = (FlightAxis *)arg;
 
 #ifndef __CYGWIN__
```

Comment:
Could we simplify this with `#ifdef __CYGWIN__` and `#elif`?

Reply:
Do you mean like this?
```
#ifdef __CYGWIN__
    //do nothing
#elif defined(__APPLE__) && defined(__MACH__)
    pthread_setname_np("ardupilot-flightaxis");
#else
    pthread_setname_np(pthread_self(), "ardupilot-flightaxis");
#endif
```

Reply:
Yes, indeed.  I think that's downright pretty compared to the nested-if.  As pretty as preprocessor stuff gets, anyway.

Only possible thing I could see is explaining why cygwin doesn't do the setname.

Your thoughts?

Reply:
Yes, seems more readable.

I don't know why cygwin doesn't support it, that was a fix pushed by @tridge in cdd350f93e55df764d2f0bc7eaa4f94ed5fa7456.

After a bit of research it just seems unsupported, but I found some interesting info in both answers here https://stackoverflow.com/questions/30346649 and also here https://stackoverflow.com/a/7989973/648044